Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/CachedQueries/PlanExtra.hs

```diff
@@ -48,6 +49,29 @@ findByMerchantOpCityIdAndPaymentModeWithServiceName (Id merchantOperatingCityId)
     Just a -> pure a
     Nothing -> cacheByMerchantIdAndPaymentMode (Id merchantOperatingCityId) paymentMode serviceName mbIsDeprecated /=<< Queries.findByMerchantOpCityIdAndPaymentModeWithServiceName (Id merchantOperatingCityId) paymentMode serviceName mbIsDeprecated
 
+------------------- -----------------------
+findByMerchantOpCityIdAndServiceName ::
+  (CacheFlow m r, MonadFlow m, EsqDBFlow m r) =>
+  Id DMOC.MerchantOperatingCity ->
+  ServiceNames ->
+  Maybe Bool ->
+  m [Plan]
+findByMerchantOpCityIdAndServiceName (Id merchantOperatingCityId) serviceName mbIsDeprecated =
+  Hedis.withCrossAppRedis (Hedis.safeGet $ makeMerchantIdAndServiceNameKey (Id merchantOperatingCityId) serviceName mbIsDeprecated) >>= \case
+    Just a -> pure a
+    Nothing -> cacheByMerchantIdAndServiceName (Id merchantOperatingCityId) serviceName mbIsDeprecated /=<< Queries.findByMerchantOpCityIdAndServiceName (Id merchantOperatingCityId) serviceName mbIsDeprecated
+
+cacheByMerchantIdAndServiceName ::
+  (CacheFlow m r) =>
+  Id DMOC.MerchantOperatingCity ->
+  ServiceNames ->
+  Maybe Bool ->
+  [Plan] ->
+  m ()
+cacheByMerchantIdAndServiceName (Id merchantOperatingCityId) serviceName mbIsDeprecated plans = do
+  expTime <- fromIntegral <$> asks (.cacheConfig.configsExpTime)
+  Hedis.withCrossAppRedis $ Hedis.setExp (makeMerchantIdAndServiceNameKey (Id merchantOperatingCityId) serviceName mbIsDeprecated) plans expTime
```

Comment:
why unwrap `Id` and then wrap again? you can pass it directly
`Hedis.setExp (makeMerchantIdAndServiceNameKey merchantOperatingCityId serviceName mbIsDeprecated) plans expTime`